File: TeleVentas/Clases/GerenteRelaciones.py
from Clases.Usuario import Usuario 
from Clases.Queja import Queja

class GerenteRelaciones(Usuario):

    def __init__(
        self,
        user_id: str,
        contrasena: str,
        nombre: str            
    )-> None:
        super().__init__(user_id,nombre,contrasena)
        self.__quejas = []


    "Metodos"

    def recibir_queja(self, queja:Queja)-> None:
        if queja.estado != Queja.EN_REVISION:
            raise ValueError(
                f"La queja no ha sido remitida aún. "
                f"Estado actual: {queja.estado}"
            )
        self.__quejas.append(queja)
        print(
            f"Gerente {self.nombre} recibió la queja {queja.numero_queja}. "
            f"Cliente: {queja.cliente.nombre}. "
            f"Motivo: {queja.motivo}"
        )
            
    def __str__(self) -> str:
        return (
            f"GerenteRelaciones(id={self.user_id}, "
            f"nombre={self.nombre}, "
            f"quejas_recibidas={len(self.__quejas)})"
        )
 
    def __repr__(self) -> str:
        return self.__str__()

File: 2.1Televentas_Java/ClasesTeleventas/GerenteRelaciones.java
package ClasesTeleventas;

import java.util.ArrayList;
import java.util.List;

/**
 * Representa al gerente de relaciones del sistema TeleVentas.
 *
 * Extiende Usuario y es responsable de recibir y gestionar
 * las quejas remitidas por los clientes.
 */
public class GerenteRelaciones extends Usuario {

    // Lista que acumula las quejas recibidas durante la sesion
    private List<Queja> quejas;

    // --- Constructor ---

    /** Inicializa el gerente con sus credenciales y una lista vacia de quejas. */
    public GerenteRelaciones(String userId, String contrasena, String nombre) {
        super(userId, nombre, contrasena);
        this.quejas = new ArrayList<>();
    }

    // --- Getters ---

    /** Retorna la lista de quejas recibidas por el gerente. */
    public List<Queja> getQuejas() {
        return quejas;
    }

    // --- Metodos ---

    /** Recibe una queja ya remitida y la agrega a la lista del gerente. */
    public void recibirQueja(Queja queja) {
        if (!queja.getEstado().equals(Queja.EN_REVISION)) {
            throw new IllegalStateException(
                "La queja no ha sido remitida aun. Estado actual: " + queja.getEstado());
        }
        quejas.add(queja);
        System.out.println("Gerente " + getNombre() + " recibio la queja " + queja.getNumeroQueja() + ". " +
                           "Cliente: " + queja.getCliente().getNombre() + ". " +
                           "Motivo: " + queja.getMotivo());
    }

    @Override
    public String toString() {
        return "  GERENTE DE RELACIONES\n" +
               "  id     : " + getUserId()   + "\n" +
               "  nombre : " + getNombre()   + "\n" +
               "  quejas : " + quejas.size() + "\n";
    }
}
